File: test/complexity.lua
local function collect(items)
    local total = 0

    for _, item in ipairs(items) do
        if item.active and item.value > 0 then
            total = total + item.value
        end
    end

    return total
end

function Example.process(input)
    if input < 0 then
        return 0
    end

    while input > 0 do
        input = input - 1
    end

    return input
end

return {
    collect = collect
}

File: test/complexity.ts
type Item = {
    active: boolean;
    value: number;
};

export function calculate(items: Item[]): number {
    let total = 0;

    for (const item of items) {
        if (item.active && item.value > 0) {
            total += item.value;
        }
    }

    return total;
}

export class TypeScriptExample {
    public transform(input: number): number {
        if (input < 0) {
            return 0;
        }

        switch (input) {
            case 1:
                return 10;
            case 2:
                return 20;
            default:
                return input;
        }
    }
}
